### DISCUSSION TITLE
Trying to detect lots of objects close together in an aerial view; railway sleepers

### DISCUSSION THREAD

**Discussion by ylankovs** [AlexeyAB/darknet#7833] [Q&A]
Hello!

We've been trying to train yolov3 tiny weights for aerial view railway sleepers. We have lots of images, split them up into 512 by 512 pixel smaller images to prevent yolo doing any image suppression.

We have tried various ways of labelling objects and training models:

1. labelling the entire sleeper with part of the rails
2. labelling the sleeper part inside the rails
3. generating extra images with changes in brightness, contrast, color and sharpness; about 10k extra images
4. trying to feed only rails that are horizontal or vertical
5. trying to feed all type of rails, horizontal, vertical, at an angel
6. running for 4k, 8k, 12k iterations
7. changing batch and subdivisions, 64/64, 24/8 etc
8. rebuilding anchors for each set of images
9. training from yolov3 tiny weights as base, continuation training from our generated weights

Train and test images split into 80/20.

mAP value doesn't go above 13%, average of 9.5%. Error value doesn't go below 4.2.

When we try detection, it seems to detect region between the sleepers, rails and sometimes sleepers. It skips a lot of the rails.

We cannot seem to get the grip on training objects like this. Some pointers to resolve our problem would be fantastic!

Note: we have trained other objects like vehicles etc without problems, but with rail sleepers there just seems to be a lot of them in the image and we either not labelling correctly or maybe the 512 by 512 frame size is too large as it contains a whole bunch of objects inside one image.

I've attached images for config file, training graph, detection results.

https://ibb.co/4tgv5tQ
https://ibb.co/2gbGp0w
https://ibb.co/hK8DcfZ
https://ibb.co/PjK7XXn
https://ibb.co/16GrcBH

**Comment by stephanecharette** [comment]
Let me know if you'd like me to take a look (stephanecharette@gmail.com).  I'd be curious to see what the problem is.  But from the description and the links you provided I cannot comment other than to say take a look at the "if the results aren't great" section here:  https://www.ccoderun.ca/programming/darknet_faq/#tldr

**Reply by ylankovs** [reply]
You have some fantastic guides and I will definitely take some of the pointers from there, and apply to our problem. I'll come back with the results in a few days. Thanks!
